library/mr_provisioner_preseed.py

```diff
@@ -77,74 +80,153 @@
 
 from ansible.module_utils.basic import AnsibleModule
 
+class ProvisionerError(Exception):
+    def __init__(self, message):
+        super(ProvisionerError, self).__init__(message)
+
+class PreseedUploader(object):
+    """ This class handles the job of uploading a preseed file to MrP.
+        It shall only be called if there is a file to be uploaded, else you can
+        fetch the thing via the regular call in the Ansible role"""
+    def __init__(self, mrp_url, mrp_token, preseed_file, preseed_name,
+                 preseed_type, preseed_desc = '', preseed_knowngood = False,
+                 preseed_public = False):
+        self.url = mrp_url
+        self.authhead = { 'Authorization': mrp_token }
+        self.file = preseed_file
+        self.name = preseed_name
+        self.type = preseed_type
+        self.id = -1
+        self.desc = preseed_desc
+        self.knowngood = preseed_knowngood
+        self.public = preseed_public
+
+    def _check_for_existence(self):
+        url = urljoin(self.url, '/api/v1/preseed?show_all=true')
+        r = requests.get(url, headers=self.authhead)
+        if r.status_code != 200:
+            raise ProvisionerError('Error fetching {}, HTTP {} {}'.format(url,
+r.status_code, r.reason))
+
+        for preseed in r.json():
+            if preseed['name'] == self.name:
+                self.id = preseed['id']
+                return True
+
+        return False
+
+    def _get_preseed_from_file(self):
+        json_preseed = {}
+
+        f = open(self.file, 'r')
+        contents = ''
+        for line in f:
+            contents += line
+        json_preseed['content'] = contents
+        json_preseed['name'] = self.name
+        json_preseed['type'] = self.type
+        json_preseed['public'] = self.public
+        json_preseed['known_good'] = self.knowngood
+        if (self.desc != ''):
+            json_preseed['description'] = self.desc
+
+        return json_preseed
+
+    def upload_preseed(self):
+        """Uploads preseed. Should check first that preseed doesn't exist, else
+        it modifies preseed (separate function ?). Post to upload, Put to
+        modify (+ id). Maybe implement a jinja2 syntax check ? But that should
+        be done on mrp's side"""
+        res = {}
+        try:
+            exists = self._check_for_existence()
+        except ProvisionerError as e:
+            res['error'] = str(e)
+            return res
+
+        if not exists and self.file == '/dev/null':
```

Comment:
Why using /dev/null as a magic string instead of '' ?

Reply:
I got your idea, but I stick to empty string.

Reply:
In the python code why not, but I stick to '/dev/null' in the YAML... right ?